### PR TITLE
feat(builds): rework build for composite projects

### PR DESCRIPTION
**Pull request by ttomsu**
From @cfieber's original commit: 

This partitions the gradle support into a dev plugin to support the ability to compile and test the project and a publishing plugin with opinions on how versioning and release management should happen.

Consuming projects will only apply the dev plugin by default, and can apply the publishing plugin via an include file to do publish to bintray and do tag based versioning.

This allows a couple things:
1. There are no external dependencies in the dev plugin, so there will not be any buildscript classpath conflicts if trying to use a project as part of a composite build with an internal build that has differing publishing opinions.

2. Release management opinions aren't applied until they are actually needed - the TravisCI build for a project can do versioning and publishing in one way, whereas the umbrella Spinnaker release process is free to do it another way.

### CODE REVIEW

**Comment by ttomsu** [comment]
Supercedes https://github.com/spinnaker/spinnaker-gradle-project/pull/39.

Ref: https://github.com/spinnaker/spinnaker/issues/2844

File: spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.ospackage.publish.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2015 Netflix, Inc.
+# Copyright 2018 Netflix, Inc.
```
Comment (dibyom) [inline spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.ospackage.publish.properties]: Probably undo the year change?

File: spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.package.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2015 Netflix, Inc.
+# Copyright 2018 Netflix, Inc.
```
Comment (dibyom) [inline spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.package.properties]: Same

File: spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.ospackage.publish.properties (hunk 1 above)
Comment (ttomsu) [reply spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.ospackage.publish.properties]: Done.

File: spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.package.properties (hunk 2 above)
Comment (ttomsu) [reply spinnaker-publishing-plugin/src/main/resources/META-INF/gradle-plugins/spinnaker.package.properties]: Done.

**Pull request merged**